pkgs/by-name/oa/oakctl/package.nix

```
@@ -0,0 +1,76 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  patchelf,
+  glibc,
+  versionCheckHook,
+}:
+
+let
+  version = "0.2.12";
+
+  # Note: Extracted from install script
+  # https://oakctl-releases.luxonis.com/oakctl-installer.sh
+  sources = {
+    x86_64-linux = fetchurl {
+      url = "https://oakctl-releases.luxonis.com/data/${version}/linux_x86_64/oakctl";
+      hash = "sha256-HCnFD0LD6sQp9k3SP2g4svjA5/kLvfrnN+IwiuMWGCY=";
+    };
+    aarch64-linux = fetchurl {
+      url = "https://oakctl-releases.luxonis.com/data/${version}/linux_aarch64/oakctl";
+      hash = "sha256-1oJQs57/tW3rsMM+LAuKiBUf1aKOKFoPQAMcVUfXqlE=";
+    };
+    aarch64-darwin = fetchurl {
+      url = "https://oakctl-releases.luxonis.com/data/${version}/darwin_arm64/oakctl";
+      hash = "sha256-arS2qfd/Z/ZCNWAKD9bc2PMwkhVtO5WViTibMST7zd8=";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://oakctl-releases.luxonis.com/data/${version}/darwin_x86_64/oakctl";
+      hash = "sha256-yyvDQbFEtlB8xmdbxquy22wAIUcCSVchP/AuSpi4TAU=";
+    };
+  };
+
+  src =
+    sources.${stdenv.hostPlatform.system}
+      or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+in
+stdenv.mkDerivation {
+  pname = "oakctl";
+  inherit version src;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgramArg = [ "version" ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -m 0755 $src $out/bin/$pname
```

```suggestion
    install -m 0755 $src $out/bin/$pname
```

While this _may_ work (I think the `pname` env variable is indeed set), it’s preferable to use string interpolation with `${pname}`. Same for the `postFixup` phase.